internal/plugin/connectors/tcp/mssql/mock/mock.go

```diff
@@ -94,3 +95,18 @@ func (n *NetConn) Write([]byte) (numBytes int, err error) {
 func FakeTdsBufferCtor(r io.ReadWriteCloser) io.ReadWriteCloser {
 	return r
 }
+
+// TdsBufferCtor returns a Buffer (with a fake transport) that can be written & read
+func TdsBufferCtor(r io.ReadWriteCloser) io.ReadWriteCloser {
+	return mssql.NewTdsBuffer(1024, new(Transport))
```

Comment:
I'm wondering if an additional Transport Type is necessary, whatever is passed to TdsBufferCtor should be the transport.

Reply:
I'm not sure I understand,  can you explain this more? We just need the mock to perform write operations without failing, we don't need an actual transport though.